subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -212,7 +212,11 @@ static int json_add_obj_cs(cJSON *parent, const char *str, cJSON *item)
 	if (!parent || !str || !item) {
 		return -EINVAL;
 	}
-	return cJSON_AddItemToObjectCS(parent, str, item) ? 0 : -ENOMEM;
+	if (!cJSON_AddItemToObjectCS(parent, str, item)) {
+		cJSON_Delete(item);
+		return -ENOMEM;
+	}
```

Comment:
I might have this backwards, but I think CJSON-like functions should only attempt to free resources that they themselves created

Since AddItemToObjectCS failed, the ownership of `item` was not transferred, and it is the original owner's responsibility to free it.

@jayteemo can you confirm my understanding of this?

Reply:
lol, looks like @jayteemo and I had the same thought simultaneously:

https://github.com/nrfconnect/sdk-nrf/pull/8304#discussion_r938080283

Reply:
fixed